source/Projectile.cpp

```diff
@@ -71,18 +72,20 @@ Projectile::Projectile(const Projectile &parent, const Point &offset, const Angl
 	cachedTarget = TargetPtr().get();
 	double inaccuracy = weapon->Inaccuracy();
 	if(inaccuracy)
-	{
 		this->angle += Angle::Random(inaccuracy) - Angle::Random(inaccuracy);
-	}
-	
-	// Revert to the velocity of the parent projectile.
-	velocity -= parent.angle.Unit() * (parent.GetWeapon().Velocity() + (parent.GetWeapon().RandomVelocity()) / 2);
-	
-	//Add the velocity of the parent and the child at the child's heading.
-	velocity += this->angle.Unit() * ((parent.GetWeapon().Velocity() + (parent.GetWeapon().RandomVelocity()) / 2) + weapon->Velocity() + (Random::Real() * weapon->RandomVelocity()));
-	
-	// This retains the intended speeds, but sets the upper limit on how much of that speed can be redirected by a child's innacuracy stat.  High inaccuracies will look odd but low inaccuracies should be fine.
-	
+
+	// Given that submunitions inherit the velocity of the parent projectile,
+	// it is often the case that submunitions don't add any additional velocity.
+	// But we still want inaccuracy to have an effect on submunitions. Because of
+	// this, we tilt the velocity of submunitions in the direction of the inaccuracy.
+
+	// Find out how much speed the submunition is adding to itself.
+	marginalVelocity = weapon->Velocity() + Random::Real() * weapon->RandomVelocity();
+	// Get the velocity of the grandparent.
+	velocity -= parent.angle.Unit() * parent.marginalVelocity;
+	// Apply the inaccuracy to the parent's velocity as well as the submunition's velocity.
+	velocity += this->angle.Unit() * (parent.marginalVelocity + marginalVelocity);
+
```

Comment:
I like the method you've used here in comparison to my own.  However, now that you've added another class variable here and since we only need to know how much speed to back off and add back on due to the entire projectile (all parent munitions and the current submunition), we could again take `marginalVelocity` as a local variable for the sake of readability and add a line below 87 `totalProjectileVelocity = parent.totalProjectileVelocity + marginalVelocity`.

The current information represented by `marginalVelocity` is technically maintained in that you can take the difference between the value at the munition in question and it's parent to find what the munition in question is responsible for.